src/Workspaces/CSharp/Portable/Formatting/Rules/WrappingFormattingRule.cs

```diff
@@ -35,55 +36,59 @@ public override void AddSuppressOperations(List<SuppressOperation> list, SyntaxN
             }
         }
 
-        private void AddSpecificNodesSuppressOperations(List<SuppressOperation> list, SyntaxNode node)
+        private ValueTuple<SyntaxToken, SyntaxToken> GetSpecificNodeSuppressionTokenRange(SyntaxNode node)
         {
-            var ifStatementNode = node as IfStatementSyntax;
-            if (ifStatementNode != null)
+            var embeddedStatement = node.GetEmbeddedStatement();
+            if (embeddedStatement != null)
             {
-                AddSuppressWrappingIfOnSingleLineOperation(list, ifStatementNode.IfKeyword, ifStatementNode.Statement.GetLastToken(includeZeroWidth: true));
-
-                if (ifStatementNode.Else != null)
+                var firstTokenOfEmbeddedStatement = embeddedStatement.GetFirstToken(includeZeroWidth: true);
+                if (embeddedStatement.IsKind(SyntaxKind.Block))
                 {
-                    AddSuppressWrappingIfOnSingleLineOperation(list, ifStatementNode.Else.ElseKeyword, ifStatementNode.Else.Statement.GetLastToken(includeZeroWidth: true));
+                    return ValueTuple.Create(
+                        firstTokenOfEmbeddedStatement.GetPreviousToken(includeZeroWidth: true),
```

Comment:
extract `firstTokenOfEmbeddedStatement.GetPreviousToken(includeZeroWidth: true)`out of `if`